Library/Homebrew/software_spec.rb

```diff
@@ -162,12 +162,25 @@ def depends_on(spec)
     add_dep_option(dep) if dep
   end
 
-  def uses_from_macos(spec, _bounds = {})
-    spec = [spec.dup.shift].to_h if spec.is_a?(Hash)
+  def uses_from_macos(deps, bounds = {})
+    if deps.is_a?(Hash)
+      bounds = deps.dup
+      deps = [bounds.shift].to_h
+    end
+
+    @uses_from_macos_elements << deps
 
-    @uses_from_macos_elements << spec
+    # Linux simulating macOS. Assume oldest macOS version.
+    return if Homebrew::EnvConfig.simulate_macos_on_linux? && !bounds.key?(:since)
```

Comment:
This makes me wonder about whether this logic should be in `extend/os/linux`; it feels a bit weird to be introducing OS-condition logic here (and in `formula.rb`).